crates/flowctl/src/raw/suggest_schema.rs

```diff
@@ -0,0 +1,222 @@
+                }, broker::Label {
+                    name: format!("{}kind", labels::FIELD_PREFIX),
+                    value: "capture".to_string(),
+                }, broker::Label {
+                    name: format!("{}name", labels::FIELD_PREFIX),
+                    value: urlencoding::encode(task).into_owned(),
+                }]
+        }),
+        exclude: None,
+    };
+
+    let client = ctx.controlplane_client().await?;
+    let mut data_plane_client =
+        dataplane::journal_client_for(client, vec![ops_collection]).await?;
+    let log_reader = journal_reader(&mut data_plane_client, &selector).await?;
+    let mut log_stream = AsyncByteLines::new(BufReader::new(log_reader.compat())).into_stream();
+    let mut log_invalid_documents = log_stream.try_filter_map(|log| async move {
+        let parsed: Log = serde_json::from_slice(&log)?;
+        if parsed.message != "document failed validation against its collection JSON Schema" {
+            return Ok(None);
+        }
+
+        let error_string = &parsed.fields_json_map.get("error").ok_or(std::io::Error::new(ErrorKind::Other, "could not get 'error' field of ops log"))?;
+        let mut err: Vec<serde_json::Value> = serde_json::from_str(error_string)?;
+        let err_object = err.pop().ok_or(std::io::Error::new(ErrorKind::Other, "could not get second element of 'error' field in ops log"))?;
+        let failed_validation: FailedValidation = serde_json::from_value(err_object)?;
+
+        let buf: Bytes = serde_json::to_vec(&failed_validation.document)?.into();
+        return Ok(Some(buf));
+    });
+
+    let mut log_invalid_documents_reader = StreamReader::new(Box::pin(log_invalid_documents));
+
+    let codec = JsonCodec::new(); // do we want to limit length here? LinesCodec::new_with_max_length(...) does this
+    let mut doc_bytes_stream = FramedRead::new(FuturesAsyncReadCompatExt::compat(reader).chain(log_invalid_documents_reader), codec);
+
+    let schema_model = collection_def.schema.unwrap();
+
+    let mut accumulator = raw_schema_to_shape(&schema_model)?;
+    let original_schema_final = SchemaBuilder::new(accumulator.clone()).root_schema();
+
+    loop {
+        match doc_bytes_stream.next().await {
+            Some(Ok(doc_val)) => {
+                let parsed: Value = doc_val;
+                // There should probably be a higher-level API for this in `journal-client`
+
+                if parsed.pointer("/_meta/ack").is_none() {
+                    let inferred_shape = infer_shape(&parsed);
+
+                    accumulator = shape::merge(accumulator, inferred_shape)
+                }
+            }
+            Some(Err(e)) => return Err(InferenceError::from(e).into()),
+            None => break
+        }
+    }
+
+    let new_schema = serde_json::to_value(&SchemaBuilder::new(accumulator).root_schema())?;
+
+    std::fs::write("original.schema.json", serde_json::to_string_pretty(&original_schema_final)?)?;
```

Comment:
The original purpose of the files is to run diff, but I personally find it useful to be able to see the actual files if I'm using this command, to be able to better understand what is the full resulting schema, and sometimes compare it manually to the existing schema. It also sometimes makes it easier to copy the new inferred schema from this file rather than from the diff (which is mixed with `+` and `-` and other signs in between).

Regarding the naming of the files... we could name them after the collection to make it easier to keep them organised and not overwrite files frequently